R/utils.R

```diff
@@ -126,3 +126,73 @@ stop_or_reject <- function(message, async = FALSE) {
   }
   stop(err)
 }
+
+as_predicate <- function(arg, env = rlang::caller_env()) {
+  if(rlang::is_formula(arg) || is.function(arg)) {
+    fun <- rlang::as_function(arg, env = env)
+  } else {
+    fun <- function(x) identical(x, arg)
+  }
+
+  function(...) {
+    res <- fun(...)
+    if(!rlang::is_true(res) && !rlang::is_false(res)) {
```

Comment:
I thing there is a function to do this test without &&
```suggestion
    if(!rlang::is_scalar_logical(res)) {
```